### DISCUSSION TITLE
Setting battery SOC

### DISCUSSION THREAD

**Discussion by t-dan** [wills106/homeassistant-solax-modbus#985] [Q&A] [answered]
Hello,
is there a way of setting the battery SOC?
It can be set on the inverter itself as well as through the remote control (Android app).

Thank you very much,
Dan

**Comment by steps56** [comment]
Use:
number.solax_selfuse_discharge_min_soc
and
number.solax_battery_charge_upper_soc

**Reply by t-dan** [reply]
Thanks, but these entities are not shown in the HA. Although I think they should be enabled for GEN4, which I think I have (SN starts with H34A).

Did I overlook something?

Thank you.
D.

**Reply by steps56** [reply]
Look at Entities at the Integration by setting hook at "not shown", "disabled", "hidden" and enable all, resart HA and it should show up.

Same Gen4 and so on use by myself, so that's fine 👌.

As I got it in HA, you will get it too 😎👍
I will send Screens, if really necessary... but try bevor please,  as I told here.

**Reply by t-dan** [reply]
Hmm, sorry I don't see then under hidden. I even don't see e.g. "Battery State of Health" sensor as well.
There is 1 device, 255 entities reported by the integration. And only "Selfuse Backup SOC" was not enabled under controls (and a number of sensors). Ha was restarted, but still nothing ...

D.

**Reply by steps56** [reply]
![Screenshot_20240811_223440_Chrome](https://github.com/user-attachments/assets/b3f00be7-fd16-4b1d-83db-8f7254b75198)
![Screenshot_20240811_223543_Chrome](https://github.com/user-attachments/assets/60b74dc4-a56c-40e9-99a9-effbdb9f55ba)
![Screenshot_20240811_223616_Chrome](https://github.com/user-attachments/assets/8244947a-0178-43c1-af7b-487163ca18c1)

After 1. Screen above, you must restart HA !!!
Did you enable the disabled Entities bevor restarting HA ?
As you see, I am German.
If you too, lets talk German 😎

**Reply by steps56** [reply, marked as answer]
Important too:
Use the Integration Version 2024.05.3 , the last one really nice working.
Newer versions don't show some stuff and are mostly pain for me by try.
So I stay at 2024.05.3 😎👌.

**Reply by steps56** [reply]
You know how to go back/downgrade this Integration?
Hacs Config .... show last 25 Versions... HA restart... and you are able to choose 2024.05.3 👌

**Reply by t-dan** [reply]
Hello,
I have to update the HA to get the version you mentioned. However, I am not able to downgrade Solax integration - there is no "show last 25 Versions" (the oldest available is 2024.8.4).
Nevertheless, despise the issues you mentions (some sensors cannot be enabled), now I have the Solax integration where the entities are at least shown ...

Thank you.
D.

**Reply by Hetzi74** [reply]
You can downgrade to any version via Developer Tools->Actions->Select Update: Install Update->Select Device ->SolaX Inverter Modbus and then specify the version you want.
